compiler/rustc_errors/src/lib.rs

```diff
@@ -1263,6 +1263,10 @@ impl Handler {
         std::mem::take(&mut self.inner.borrow_mut().fulfilled_expectations)
     }
 
+    pub fn insert_fulfilled_diagnostic(&self, expectation_id: LintExpectationId) {
+        self.inner.borrow_mut().fulfilled_expectations.insert(expectation_id);
+    }
```

Comment:
Before this PR the only way to insert into the `fulfilled_expectations` was via the `emit_diagnostics` method https://github.com/rust-lang/rust/blob/81c650fd69886441d52c201681f639609de4640f/compiler/rustc_errors/src/lib.rs#L1351-L1354

I haven't yet poked around to try to figure out whether or not exposing a way to insert without emitting a diagnostic makes it possible to break some important invariant.

Tips appreciated.